### PR TITLE
[multistage][bugfix] Fix not-in clause not execute properly

### LINKED ISSUE #9867: [multistage] Add Support for Not-In Operator
At present the following query doesn't work:

```
SELECT
  teamID
FROM
  baseballStats
WHERE
  teamID NOT IN (
    SELECT
      teamID
    FROM
      baseballStats
    WHERE
      hits > 10
  )
```

This gets converted to a left-join which is correct. However there a couple of issues:

1. The FilterOperand doesn't recognize the "IS NOT TRUE" function.
2. Projection for the table-scan is not pushed down, so the join ends up pulling in data for all the columns.

re: 2, it also hits a bug where the mailbox send operator receives a block which has 1 less column (teamID0 column) than expected.

This query can also benefit from colocated joins.

cc: @walterddr

### PR DESCRIPTION
**Pull request by walterddr**
Summary
===
- this is a follow up on https://github.com/apache/pinot/pull/9886
- this fixes #9867

Changelist
===
1. FilterOperand now support ISTRUE and ISNOTTRUE, which can be generated by Calcite
2. FilterOperand now use BooleanUtils.toBoolean to convert to boolean (not ideal in performance but surely for correctness)
3. Changed conversion rule for Boolean type in DataSchema b/c MIN(boolCol) and MAX(boolCol) returns a DOUBLE instead of an INT

### CODE REVIEW

**Comment by walterddr** [comment]
@ankitsultana let me know if anything else you would like me to test

**Review by ankitsultana — APPROVED** [review]

**Comment by ankitsultana** [comment]
> @ankitsultana let me know if anything else you would like me to test

@walterddr : There was also a bug where the projection wasn't pushed down correctly (table scan node was returning all columns to the join stage). I guess we'll take that up in a different PR?

File: pinot-query-runtime/src/test/resources/queries/TableExpressions.json [hunk 1]
```diff
@@ -16,6 +16,9 @@
       }
     },
     "queries": [
+      { "sql": "SELECT * FROM {tbl} WHERE strCol IN (SELECT strCol FROM {tbl} WHERE intCol < 100)" },
```
Comment (siddharthteotia) [inline pinot-query-runtime/src/test/resources/queries/TableExpressions.json:19]: may be consider adding a test for float / double as well and test for both `zero point something` and `nonzero point something` if not already covered ?

File: pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java [hunk 2]
```diff
@@ -456,7 +460,7 @@ public Serializable convertAndFormat(Object value) {
         case BIG_DECIMAL:
           return (BigDecimal) value;
         case BOOLEAN:
-          return (Integer) value == 1;
+          return ((Number) value).intValue() == 1;
```
Comment (siddharthteotia) [inline pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java:463]: may want to double check this like discussed offline

**Review by siddharthteotia — APPROVED** [review]

File: pinot-query-runtime/src/test/resources/queries/TableExpressions.json (hunk 1 above)
Comment (walterddr) [reply pinot-query-runtime/src/test/resources/queries/TableExpressions.json:19]: actually IN cannot produce a double or a float. this will be address separately
